executor/insert_common.go

```diff
@@ -200,13 +209,20 @@ func insertRows(ctx context.Context, base insertCommon) (err error) {
 	batchInsert := sessVars.BatchInsert && !sessVars.InTxn()
 	batchSize := sessVars.DMLBatchSize
 
+	// rowIdx and lazyFillAutoID are correlated.
+	e.rowIdx = -1
+	e.lazyFillAutoID = true
```

Comment:
do we need to make it configurable mode like `innodb`， to meet different needs like performance and consecutive needs

Reply:
innodb_autoinc_lock_mode is not  temporary suitable for TiDB now. 
eg：
when do `insert select from` in mode = 1(consecutive mode), innodb can guarantee the consecutive autoid using table_level lock until the end of the statement, but tidb can't guarantee the preserved autoid batch can cover the unknown row flow. So the mode distinction is not suitable.

now tidb only supports consecutive autoid in an insert batch (batch rows only may allocate once), differing from eval row (allocate multiple times). So it's not a performance compromise.

Reply:
`lazyFillAutoID` is always `true`? Then why we need it?

Reply:
> `lazyFillAutoID` is always `true`? Then why we need it?

Now we only support consecutive autoid in `insert into values()()` statement. 
Other statements still use traditional autoID allocation.(allocated one for one row).